### ISSUE TITLE
High CPU Usage

### ISSUE THREAD

**Issue by jc260552** [ipsingh06/seedsync#11]
Hi, I just discovered seedsync and it is exactly what I needed. 
However it seems to be very cpu heavy.

I monitored CPU usage overnight (while not downloading) and it constantly hovers at around ~50%
![seedsynccpuusage](https://user-images.githubusercontent.com/8284764/39661410-40b544fa-5090-11e8-9d92-fbe1b6e94bb1.png)

I have set the scan intervals to 2mins as a sanity check. But it seems to have no effect.
![seedsyncsettings](https://user-images.githubusercontent.com/8284764/39661497-832fee56-5091-11e8-9937-d5b7e5d52b33.png)

**Comment by jc260552** [comment]
Doing a perf doesn't seem to report anything useful.
```
Samples: 10K of event 'cycles', Event count (approx.): 8928436057
Overhead  Command   Shared Object         Symbol
  20.41%  seedsync  libpython3.5m.so.1.0  [.] PyEval_EvalFrameEx
   5.01%  seedsync  libpython3.5m.so.1.0  [.] _PyObject_GenericGetAttrWithDict
   3.37%  seedsync  libpython3.5m.so.1.0  [.] _PyType_Lookup
   2.04%  seedsync  libpython3.5m.so.1.0  [.] PyDict_GetItem
   1.77%  seedsync  libpython3.5m.so.1.0  [.] PyFrame_New
   1.59%  seedsync  libpython3.5m.so.1.0  [.] PyObject_GetAttr
   1.20%  seedsync  libpython3.5m.so.1.0  [.] PyObject_RichCompare
   1.07%  seedsync  libpython3.5m.so.1.0  [.] PyObject_Call
   1.06%  seedsync  libpython3.5m.so.1.0  [.] _Py_CheckFunctionResult
   0.95%  seedsync  libpython3.5m.so.1.0  [.] PyUnicode_RichCompare
   0.91%  seedsync  libpython3.5m.so.1.0  [.] PyTuple_New
   0.79%  seedsync  libpython3.5m.so.1.0  [.] 0x00000000000beeda
   0.78%  seedsync  libpython3.5m.so.1.0  [.] _PyDict_LoadGlobal
   0.76%  seedsync  libpython3.5m.so.1.0  [.] PyType_IsSubtype
   0.75%  seedsync  libc-2.24.so          [.] _IO_vfscanf
   0.68%  seedsync  libc-2.24.so          [.] malloc
   0.67%  seedsync  libpython3.5m.so.1.0  [.] 0x000000000022eab4
   0.59%  seedsync  [kernel.kallsyms]     [k] ___bpf_prog_run
   0.58%  seedsync  libpython3.5m.so.1.0  [.] PyObject_GC_UnTrack
   0.53%  seedsync  libpython3.5m.so.1.0  [.] PyErr_Occurred
   0.44%  seedsync  [kernel.kallsyms]     [k] __d_lookup_rcu
   0.44%  seedsync  libpython3.5m.so.1.0  [.] 0x000000000009f710
   0.42%  seedsync  libpython3.5m.so.1.0  [.] 0x000000000022ea14
   0.41%  seedsync  libpython3.5m.so.1.0  [.] PyDict_SetItem
   0.40%  seedsync  libpython3.5m.so.1.0  [.] 0x00000000000c2c04
   0.39%  seedsync  libpython3.5m.so.1.0  [.] 0x000000000009f723
   0.38%  seedsync  libpython3.5m.so.1.0  [.] _PyDict_Next
```

I can't seem to get any python profilers to work

**Comment by flatlinebb** [comment]
I am also experiencing the high CPU usage bug. Is there any progress on addressing this bug? Is there any info I can provide that can help? Running the docker instance.

**Comment by restalfep** [comment]
I'm also experiencing this. It has made this script unusable. I noticed that when certain characters it also seems worse (like if Japanese characters and such). Some issue with ASCII it says.

**Comment by ipsingh06** [comment]
This has now been fixed in v0.7.0. Thank you for reporting this issue!
